### PR TITLE
Feature/cms 1211 radio btn group

### PR DESCRIPTION
**Pull request by VerenaNetlight**
- Maybe more style changes are needed to better fit the other components, @ornellagiau ?
- _Normal_ & _Materia_ theme are not specifically set. Do they need to be implemented and if yes, how should they look like @sagold ?
- There's currently no focus shadow implemented (as those radio buttons are only pseudo radio buttons). Should this be changed?

### CODE REVIEW

File: components/radiobuttons/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                oncreate: _vnode => (this.$container = _vnode.dom as HTMLElement),
+            },
+            options.map((option: OptionValue) => {
+                const icon = option.icon ? 
+                    m("span.mmf-icon", {  
+                        value: option.value,
+                    }, option.icon) 
+                    : undefined;
+
+                return m(".mmf-radio-btn",
```
Comment (sagold) [inline components/radiobuttons/index.ts:40]: 1. using divs here will prevent tabbing through form-elements for this kind of input-type. We should use html-buttons here, which support states like focus and tab-index
2. using html-buttons its getting obvious that we can reuse the button component to share styles, layout and behaviour
3. introducing buttons as form-element here, will add support for keyboard input

File: components/radiobuttons/index.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    {
+                        class: `${option.value === value ? "selected" : ""}`,
+                        value: option.value,
+                        disabled: disabled ? disabled : option.disabled || false,
+                        onclick: event => {
+                            event.target.focus();
+                            if ( disabled || option.disabled) return;
+                            Array.from(this.$container?.children).forEach(el => el.classList.remove("selected"));
+                            event.target?.parentNode?.classList?.add("selected");
```
Comment (sagold) [inline components/radiobuttons/index.ts:49]: grabbing through dom is prone to errors. We could store our required dom-elements in oncreate-lifecycle hook

File: components/radiobuttonsform/index.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import m from "mithril";
+import RadioButtons, { Attrs as RadioButtonsAttrs } from "../radiobuttons";
+import Label from "../label";
+import Errors, { getErrorClass } from "../errors";
+import { DefaultFormAttrs } from "../types";
+
+
+export const defaultOptions = {
```
Comment (sagold) [inline components/radiobuttonsform/index.ts:8]: Same here. For consistency use `DefaultFormAttrs & InputAttrs & { customAttributes }`

File: components/radiobuttons/layout.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        color: map-get($color-highlight, 0);
+    }
+
+    &.selected {
+        border: $mmf-input-border--width solid map-get($color-highlight, 5);
+        background-color:map-get($color-highlight, 5);
+        border-radius: $mmf-radius;
+        color: map-get($color-highlight, 0);
+    }
+}
```
Comment (sagold) [inline components/radiobuttons/layout.scss:47]: By convention we usually keep a last empty line within each file to maintain support for line-based shell-tools

**Review by sagold — CHANGES_REQUESTED** [review]

File: components/radiobuttons/layout.scss (hunk 4 above)
Comment (sagold) [reply components/radiobuttons/layout.scss:47]: It seems you dont have a plugin for .editorconfig files. There you can find the option for all files:

```
insert_final_newline = true
```

File: app/index.ts [hunk 5]
```diff
@@ -121,6 +119,222 @@ const template = {
...
+            },
+            "disabled radio elements without icons": {
+                attrs: {
+                    value: "wine",
+                    options: [
+                        {
+                            id: "coffee-no-icon-disabled",
+                            title: "Coffee", 
+                            value: "coffee",
+                            disabled: true
```
[the remark this answers is not in the document]
Comment (VerenaNetlight) [reply app/index.ts:222]: There is an optional one on the value level and one optional on the options level, to also enable disable of single options. What do you think?

File: components/radiobuttons/index.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                    }, option.icon) 
+                    : undefined;
+
+                return m(".mmf-radio-btn",
+                    {
+                        class: `${option.value === value ? "selected" : ""}`,
+                        value: option.value,
+                        disabled: disabled ? disabled : option.disabled || false,
+                        onclick: event => {
+                            event.target.focus();
```
[the remark this answers is not in the document]
Comment (VerenaNetlight) [reply components/radiobuttons/index.ts:46]: nope, will remove it

File: components/radiobuttons/index.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                            if ( disabled || option.disabled) return;
+                            Array.from(this.$container?.children).forEach(el => el.classList.remove("selected"));
+                            event.target?.parentNode?.classList?.add("selected");
+                            vnode.attrs.onchange(event.target.value || event.target.getAttribute("value"));
```
[the remark this answers is not in the document]
Comment (VerenaNetlight) [reply components/radiobuttons/index.ts:50]: true, forgot to change this after removing the radio inputs 👍🏾

File: app/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (sagold) [reply app/index.ts:222]: so this is intentional. great!

**Pull request merged**